tailscale/data_source_user.go

```diff
@@ -0,0 +1,119 @@
+package tailscale
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	tsclient "github.com/tailscale/tailscale-client-go/v2"
+)
+
+var userSchema = map[string]*schema.Schema{
+	"id": {
+		Type:        schema.TypeString,
+		Description: "The unique identifier for the user.",
+		Required:    true,
+	},
+	"display_name": {
+		Type:        schema.TypeString,
+		Description: "The name of the user.",
+		Computed:    true,
+	},
+	"login_name": {
+		Type:        schema.TypeString,
+		Description: "The emailish login name of the user.",
+		Computed:    true,
+	},
```

Comment:
I think it might make sense to allow for optionally specifying the `display_name` or `login_name` for specifying the user in addition to the id for better ergonomics / readability of the resulting Terraform code. This would be similar to  how we allow [name or hostname](https://github.com/tailscale/terraform-provider-tailscale/blob/main/tailscale/data_source_device.go#L20-L30) for finding a device.

On a related note: do we currently have the user ID surfaced somewhere in the UI? Or do people need to hit the API to get this information

Reply:
We do not surface the ID